### ISSUE TITLE
 Error polling for event status: failed to query event: CUDA_ERROR_UNKNOWN during training process

### ISSUE THREAD

**Issue by cristiangogosila** [tensorflow/tensorflow#27281]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 1.10
- Python version: 3.6.7
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 9.0/7.2.1
- GPU model and memory: 3x NVIDIA RTX 2080 Ti

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**
Sometimes, during training process , i get this error:  E tensorflow/stream_executor/cuda/cuda_event.cc:48] Error polling for event status: failed to query event: CUDA_ERROR_UNKNOWN: unknown error
2019-03-28 20:00:09.965396: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:274] Unexpected Event status: 1

This happens sometimes when i am not connected to that machine. For some reason, when i am working on that machine, i'm not getting that error.

**Describe the expected behavior**

**Small snippet of code
model.compile(loss='binary_crossentropy',
              optimizer='adam', metrics=['accuracy','binary_crossentropy'])
filepath="weights-{epoch:02d}-{acc:.4f}.h5"
mc = ModelCheckpoint(filepath, monitor='val_loss', verbose=0, save_best_only=False, save_weights_only=False, mode='auto', period=50)
baseline = model.fit(X,Y,epochs=200, batch_size=32,validation_data=(Xtest, Ytest), callbacks=[mc])

**Code to reproduce the issue**
It reproduces at an epoch numeber bigger than 150( sometimes is around 200, sometimes 500)

**Other info / logs**.

**Comment by sanjoy** [comment]
> This happens sometimes when i am not connected to that machine. For some reason, when i am working on that machine, i'm not getting that error.

Are you sure this is a problem with TensorFlow?  This aspect of the problem makes it sound like a systemic issue (e.g. perhaps disconnecting from the machine shuts down the GPU?).

**Comment by QuintonDenman** [comment]
I have just encountered the same bug whilst training a custom GAN built with the tensorflow framework. It also occured after a few hundred epochs and the loss function+optimiser I used are the same as christiangogosilas. Rtx 2070super, python 3.7.4, tensorflow installed using anaconda.

**Comment by sanjoy** [comment]
> I have just encountered the same bug whilst training a custom GAN built with the tensorflow framework. It also occured after a few hundred epochs and the loss function+optimiser I used are the same as christiangogosilas. Rtx 2070super, python 3.7.4, tensorflow installed using anaconda.

Did you notice anything suspicious in the logs?  If you link to them from here I'm happy to take a look as well.

**Comment by QuintonDenman** [comment]
Thank you for the offer sanjoy. Do you know if tensorflow save error logs like this somewhere? I found it hard to search for an answer online as anything with tensorflow and logs brings up code guidelines as you can imagine.

**Comment by sanjoy** [comment]
> Thank you for the offer sanjoy. Do you know if tensorflow save error logs like this somewhere?

Just what it prints out to stderr.

For extra points you could also set the environment variable `TF_CPP_MIN_VLOG_LEVEL` to `1` (used [here](https://github.com/tensorflow/tensorflow/blob/8191585626f438c185ce608e177a8c33fb9d3bee/tensorflow/core/platform/default/logging.cc#L178)) which will turn on verbose logging.  However, I'm not  sure if the logging output will remain small enough to be uploaded with this set.

**Comment by QuintonDenman** [comment]
I can try catch the print out, but unfortunately it freezes my pc then the system reboots when it does occur. I guess the GPU crashes? I do not have this problem on every run so it is even harder to catch.

**Comment by sanjoy** [comment]
> unfortunately it freezes my pc then the system reboots when it does occur. I guess the GPU crashes?

That sounds like a kernel / driver issue to me, not something specific to TF (IIUC TF should not be able to crash the machine).  Can you bring this up on NVIDIA's support forums?

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, Please open a new issue for any help you need against 2.x, and we will get you the right help.
